File: rb/features.rb
# Sharedmobilitych SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SharedmobilitychFeatures
  def self.make_feature(name)
    case name
    when "base"
      SharedmobilitychBaseFeature.new
    when "test"
      SharedmobilitychTestFeature.new
    else
      SharedmobilitychBaseFeature.new
    end
  end
end

File: py/sharedmobilitych_sdk/features.py
# Sharedmobilitych SDK feature factory

from sharedmobilitych_sdk.feature.base_feature import SharedmobilitychBaseFeature
from sharedmobilitych_sdk.feature.test_feature import SharedmobilitychTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SharedmobilitychBaseFeature(),
        "test": lambda: SharedmobilitychTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
